test/constraints-validation.test.ts

```diff
@@ -0,0 +1,73 @@
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '../lib';
+import { BlueprintBuilder, ControlPlaneLogTypes } from '../lib';
+import { z, ZodError } from 'zod';
+
+const longName = 'eks-blueprint-with-an-extra-kind-of-very-long-name-that-is-definitely-not-going-to-work';
+
+const app = new cdk.App();
+
+const addOns: Array<blueprints.ClusterAddOn> = [
+  new blueprints.addons.MetricsServerAddOn(),
+  new blueprints.addons.ClusterAutoScalerAddOn(),
+  new blueprints.addons.VpcCniAddOn(),
+];
+
+function getConstraintsDataSet(): [BlueprintBuilder, ZodError][] {
```

Comment:
Let's extract the tuple to a type to make it cleaner, the way I showed initially, e.g. 

```
type BlueprintBuilderError = [ BlueprintBuilder, ZodError ];

function getConstraintsDataSet(): BlueprintError[] { ...

Reply:
Oh yes that is a cleaner way to do it.